### ISSUE TITLE
Docking connctor blocks movement while being connected

### ISSUE THREAD

**Issue by BigBadKangaroo** [tsunamayo/Starship-EVO#5856]
Hi there,

I have a ship with a docking connector and an entity on the ground with a docking connector, that I want to dock at.
When flying up to the entity on the ground and activating the docking connector on the ship to dock with the entity on the ground, my ship is locked completely out of any movement while being docked.

![20240703214051_1](https://github.com/tsunamayo/Starship-EVO/assets/81495956/e0771273-f097-403f-b79c-5f50a652c241)

Now I'm asking myself if that is the intended behaviour? Or am I using docking connector in a way that is not supposed to be used? 
Is it just used for ground bases / space stations where you're not supposed to move? 

Because what I'm expecting in this moment is that I will be able to dock to another entity and carry it with me while being docked. Pretty much just like when using maglocks (but having a shared inventory/ fuel)
I think SpaceEngineers does is like that ... and it's pretty much the first way I thought of using the docking connector aswell.

*Sorry if that has already been brought up before, I looked it up and didn't find anything about that specific use case*

**Comment by tsunamayo** [comment]
@BigBadKangaroo yeah unfortunately it is not done for that. The way it works currently is that there is parent / children relationship, so the children cannot drag the parent.
It would require a complete refactor and a big change on the physics engine to work.
I guess what Space Engineers is doing is using a join constrain, (which is how it would work in real life). The problem for me is we go at crazy speed much higher than Space Engineers, so the physics engine would bug out if I do that (I also need the dev for join).

Alternatively I could simply say the parent is the one that initiate the docking, that would solve your issue.

**Comment by Tackietacktack** [comment]
Ah, I had the same, now I understand the problem.

I don't know if it is easy and possible but what is if you make  something like a switch
  "parents initiate docking" / "children initiate docking"?

**Comment by BigBadKangaroo** [comment]
> @BigBadKangaroo yeah unfortunately it is not done for that. The way it works currently is that there is parent / children relationship, so the children cannot drag the parent. It would require a complete refactor and a big change on the physics engine to work. I guess what Space Engineers is doing is using a join constrain, (which is how it would work in real life). The problem for me is we go at crazy speed much higher than Space Engineers, so the physics engine would bug out if I do that (I also need the dev for join).
> 
> Alternatively I could simply say the parent is the one that initiate the docking, that would solve your issue.

This is good to know, I'll go for a maglock design instead. But changeing the parent and the children entity might work well for general use ingame.

**Comment by drimmel247** [comment]
> @BigBadKangaroo yeah unfortunately it is not done for that. The way it works currently is that there is parent / children relationship, so the children cannot drag the parent. It would require a complete refactor and a big change on the physics engine to work. I guess what Space Engineers is doing is using a join constrain, (which is how it would work in real life). The problem for me is we go at crazy speed much higher than Space Engineers, so the physics engine would bug out if I do that (I also need the dev for join).
> 
> Alternatively I could simply say the parent is the one that initiate the docking, that would solve your issue.

I have a question about the different use cases for when someone should be using the shipyard versus using the connector. I like that shipyard allows any ships as long as it fits. However I want to transfer fuel and supplies which means that a connector would be needed. Is it possible for us to use connector across sizes like a ship with 1m and a station with 2m connector.
